src/interface/state_descriptor.hpp

```diff
@@ -406,6 +406,10 @@ class StateDescriptor {
     if (InitNewlyAllocatedVarsBlock != nullptr) return InitNewlyAllocatedVarsBlock(rc);
   }
 
+  void UserWorkBeforeLoop(Mesh *pmesh, ParameterInput *pin, SimTime &tm) const {
+    if (UserWorkBeforeLoopMesh != nullptr) return UserWorkBeforeLoopMesh(pmesh, pin, tm);
+  }
```

Comment:
This copies the design of the other function hooks in state descriptor, but we could cut out the middle and just check for `nullptr` in the function pointer itself.